vault/logical_system.go

```diff
@@ -224,6 +225,27 @@ type SystemBackend struct {
 	logger    log.Logger
 }
 
+// handleConfigStateSanitized returns the current configuration state. The configuration
+// data that it returns is a sanitized version of the combined configuration
+// file(s) provided.
+func (b *SystemBackend) handleConfigStateSanitized(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	config := b.Core.SanitizedConfig()
+
+	configMap := structs.New(config).Map()
```

Comment:
Don't love the fact that we need to use `structs` and then pull out values that end in `Raw`. Seems fragile if new values are added to the config struct that don't happen to end in Raw. I wonder if instead we can have `b.Core.SanitizedConfig()` return a map instead of a config object?

Reply:
We could have `SanitizedConfig()` return a map instead, but we'd still have the same problem of empty `Raw*` values in the map that will show up in the response so we'd just be removing them there as opposed to here.

Alternatively, we could have `structs:"-"` tags added to the raw fields in `server.Config`. However, I could see a case were we might want to have these show up in the future, so I'm not sure if always omitting them is the right call either.

Reply:
I've updated it to return a map directly as discussed offline via 26fdcb5def42ee4002d2af1833fe72eb0dccf1ac. There's room for improvement on how we can curate this better, but this should give us something to work with for debug.